drivers/netdev_ieee802154_submac/netdev_ieee802154_submac.c

```diff
@@ -291,8 +291,24 @@ static int _init(netdev_t *netdev)
         (netdev_ieee802154_submac_t *)netdev;
     ieee802154_submac_t *submac = &netdev_submac->submac;
     netdev_ieee802154_t *netdev_ieee802154 = (netdev_ieee802154_t *)netdev;
+    netdev_ieee802154_setup(netdev_ieee802154);
+
     ieee802154_submac_init(submac, (network_uint16_t*) netdev_ieee802154->short_addr, (eui64_t*) netdev_ieee802154->long_addr);
 
+    /* This function already sets the PAN ID to the default one */
+    netdev_ieee802154_reset(netdev_ieee802154);
+
+    uint16_t chan = CONFIG_IEEE802154_DEFAULT_CHANNEL;
+    int16_t tx_power = CONFIG_IEEE802154_DEFAULT_TXPOWER;
```

Comment:
```suggestion
    uint16_t chan = submac->channel_num;
    int16_t tx_power = submac->tx_pow;
```

Maybe get it from the submac values? so we don't lose sync between netdev and submac in case a change is made to `ieee802154_submac_init`.

Reply:
you are right, we should definitely do that.